### ISSUE TITLE
Text styled as headings are not headings

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15289]
Text styled as headings are not headings

-   **Severity:** Low
-   **Affected Populations:**
    -   Blind
    -   Cognitively Impaired
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Block Panel
    -   Title Block

#### Issue description

At the top of the Block panel, the name of the block type is styled as a
heading, and functions as a heading for the content describing the block
type. It has a class name suggesting it is meant as a section title.

In some places in the Block panel and in the Title Block, a label-input
pair is imitated with a paragraph-result pair. There is no semantic
relationship between the paragraph and the content following it.

Headings provide important wayfinding points for assistive technology
users, and help to provide context for the content that follows them.
When headings are missing or improperly defined, it may be more
difficult for such users to understand and navigate the content.

##### Issue Code

```
    <div class="editor-block-inspector__card-title">Paragraph</div>


    ...


    <div class="components-panel__body is-opened">


        ...


        <label...>URL</label>


        <input ...>


        ...


        <p class="edit-post-post-link__preview-label">Preview</p>


        <a target="_blank" class="..." href="...">...</a>


        ...


    </div>
```

#### Remediation Guidance

Ensure elements presented and styled as headings are, in fact,
headings.

In places where the format is switching between a label-input pair to a
label-result pair, use headings to semantically represent that the label
text is giving context to the content which follows it.

##### Recommended Code

```
    <h2 class="editor-block-inspector__card-title">Paragraph</h2>


    ...


    <div class="components-panel__body is-opened">


        ...


        <label...>URL</label>


        <input ...>


        ...


        <h3  class="edit-post-post-link__preview-label">Preview</h3>


        <a target="_blank" class="..." href="...">...</a>


        ...


    </div>
```

#### Relevant standards

-   [1.3.1 Info and Relationships
    (Level A)](https://www.w3.org/TR/WCAG20/#content-structure-separation-programmatic)
-   [2.4.6 Headings and Labels (Level
    AA)](https://www.w3.org/TR/WCAG20/#navigation-mechanisms-descriptive)
-   [4.1.2 Name, Role, Value
    (Level A)](https://www.w3.org/TR/WCAG20/#ensure-compat-rsv)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-35 in Tenon's report

**Comment by StommePoes** [comment]
Screenshot from the report:
![Cropped screenshot of the Block menu with "Paragraph" highlighted and block-type description underneath.](https://user-images.githubusercontent.com/136210/57184208-50ca8c00-6eb8-11e9-8df7-2ede9485e52e.png)

**Comment by afercia** [comment]
Based on the CSS class in the report, the second part of this issue relates to the Permalink panel in the  Sidebar:

<img width="320" alt="Screenshot 2019-05-05 at 23 55 23" src="https://user-images.githubusercontent.com/1682452/57200899-7081b300-6f91-11e9-920c-5d53a3b4a3b8.png">
